File: Lua/greekletters.lua
function Greek(n)
local Letter
	if (n == 1)
	then
		Letter = "Alpha";
	elseif (n == 2)
	then
		Letter = "Beta";
	elseif (n == 3)
	then
		Letter = "Gamma";
	elseif (n == 4)
	then
		Letter = "Delta";
	elseif (n == 5)
	then
		Letter = "Epsilon";
	elseif (n == 6)
	then
		Letter = "Zeta";
	elseif (n == 7)
	then
		Letter = "Eta";
	elseif (n == 8)
	then
		Letter = "Theta";
	elseif (n == 9)
	then
		Letter = "Iota";
	elseif (n == 10)
	then
		Letter = "Kappa";
	elseif (n == 11)
	then
		Letter = "Lambda";
	elseif (n == 12)
	then
		Letter = "Mu";
	elseif (n == 13)
	then
		Letter = "Nu";
	elseif (n == 14)
	then
		Letter = "Xi";
	elseif (n == 15)
	then
		Letter = "Omicron";
	elseif (n == 16)
	then
		Letter = "Pi";
	elseif (n == 17) 
	then
		Letter = "Rho";
	elseif (n == 18) 
	then
		Letter = "Sigma";
	elseif (n == 19) 
	then
		Letter = "Tau";
	elseif (n == 20) 
	then
		Letter = "Upsilon";
	elseif (n == 21) 
	then
		Letter = "Phi";
	elseif (n == 22) 
	then
		Letter = "Chi";
	elseif (n == 23) 
	then
		Letter = "Psi";
	elseif (n == 24) 
	then
		Letter = "Omega";
	elseif (n == 0)
	then
		Letter = "";
	else
		Letter = "ERROR";
	end
	return Letter
end

local Sec_1, Sec_2, Sec_3, Sec_4, Sec_5 = 0,0,0,0,0
local Number = 0
local Letter_1, Letter_2, Letter_3, Letter_4, Letter_5 = "","","","",""

print("Please enter a number, note that it must be less than 8,308,824 as that is the highest number current programming can handle\n")
print("Enter Number: ")
Number = io.read("*n")     -- read a number
--print("\n\n")

if (Number > 8308824)
then
	print("The number you entered is too large")
	print("\nIt will be reduced to 8,308,824 which is the largest number")
	Number = 8308824
end

--print("\n\n")

if (Number > 24)
then
	Sec_2 = math.floor((Number - 1) / 24)
	Sec_1 = Number - (Sec_2 * 24)
else
	Sec_1 = Number
end

if (Sec_2 > 24)
then
	Sec_3 = math.floor((Sec_2 - 1) / 24)
	Sec_2 = Sec_2 - (Sec_3 * 24)	
end

if (Sec_3 > 24)
then
	Sec_4 = math.floor((Sec_3 - 1) / 24)
	Sec_3 = Sec_3 - (Sec_4 * 24)
end

if (Sec_4 > 24)
then
	Sec_5 = math.floor((Sec_4 - 1) / 24);
	Sec_4 = Sec_4 - (Sec_5 * 24);
end

Letter_1 = Greek(Sec_1)
Letter_2 = Greek(Sec_2)
Letter_3 = Greek(Sec_3)
Letter_4 = Greek(Sec_4)
Letter_5 = Greek(Sec_5)


print(Letter_5," ",Letter_4," ",Letter_3," ",Letter_2," ", Letter_1)

File: Java/GreekLetters.java
//tutorial http://www.tutorialspoint.com/java/index.htm

/*
enum ex 
enum FreshJuiceSize {SMALL, MEDIUM, LARGE }
FreshJuiceSize size;
*/

/* 
Making object
class FreshJuice {BLAH}
juice = new Freshjuice();
*/

/*
Access control modifiers:

Visible to the package (nothing)
Visible to class only (private)
Visible to world (public)
Visible to package and all subclasses (protected)


Non Access Modifiers
static for creating class method and variables
final for finalizing the implementations of classes, methods, and variables
abstract for creating abstract classes and methods
synchronized and volatile used for threads

*/

//Now on to the actual code.

import java.io.*; //showing stuff to user
import java.util.Date; //To display the date (pointless, just testing this)
import java.util.Scanner; //for getting input from user

class Letter {
	int num;
	String Letter;
	
	//Constructor without parameter in case of error.
	public Letter() { this.num = 25; }
	
	//Constructor with parameter as it should be.
    public Letter(int number) { this.num = number;}
	
	public void getLetter() {
		switch( this.num ) {
			case 0:
				this.Letter = "";
				break;
			case 1:
				this.Letter = "Alpha";
				break;
			case 2:
				this.Letter = "Beta";
				break;
			case 3:
				this.Letter = "Gamma";
				break;
			case 4:
				this.Letter = "Delta";
				break;
			case 5:
				this.Letter = "Epsilon";
				break;
			case 6:
				this.Letter = "Zeta";
				break;
			case 7:
				this.Letter = "Eta";
				break;
			case 8:
				this.Letter = "Theta";
				break;
			case 9:
				this.Letter = "Iota";
				break;
			case 10:
				this.Letter = "Kappa";
				break;
			case 11:
				this.Letter = "Lambda";
				break;
			case 12:
				this.Letter = "Mu";
				break;
			case 13:
				this.Letter = "Nu";
				break;
			case 14:
				this.Letter = "Xi";
				break;
			case 15:
				this.Letter = "Omicron";
				break;
			case 16:
				this.Letter = "Pi";
				break;
			case 17:
				this.Letter = "Rho";
				break;
			case 18:
				this.Letter = "Sigma";
				break;
			case 19:
				this.Letter = "Tau";
				break;
			case 20:
				this.Letter = "Upsilon";
				break;
			case 21:
				this.Letter = "Phi";
				break;
			case 22:
				this.Letter = "Chi";
				break;
			case 23:
				this.Letter = "Psi";
				break;
			case 24:
				this.Letter = "Omega";
				break;
			default :
				this.Letter = "ERROR";
				break;
		} //end switch
	} //end getLetter
	
	public void showLetter() {
		System.out.print(" " + this.Letter );
	}
}


public class GreekLetters {
	
	//Some variables that really mean nothing, but I have them here to play with
	static String strVersion = "v1.0.0";
	static Date date = new Date();
	
	public static void main( String []args ) {
		System.out.println( date.toString() );
		System.out.println("Running version number: " + strVersion);
		Scanner reader = new Scanner( System.in );
		
		//prompt user for number, keep prompting if it is not in correct range		
		int input = 0;
		do {
			if (input < 0 || input > 8308824) {
				System.out.println("You have entered an incorrect number, it has to be between 0 and 8,308,824");
			}
			
			System.out.print("Please enter a number: ");
			input = reader.nextInt();
			System.out.print("\n");
		} while (input < 0 || input > 8308824);
		
		int[] sec_nums = {0, 0, 0, 0, 0}; //array of section numbers
		sec_nums[0] = input;
		
		Letter[] letterArray = new Letter[5]; //an array to hold the letter objects
		
		for (int i = 0; i < sec_nums.length; i++) {
			//the i != ( sec_nums.length - 1) is to make sure that if we are on the 
			//last array element we aren't trying to access the next element (which doesn't exist)
			if ( sec_nums[i] > 24 && i != ( sec_nums.length - 1) ) {
				sec_nums[i + 1] = (int) Math.floor( ( sec_nums[i] - 1)  / 24);
				sec_nums[i] -= sec_nums[i + 1] * 24;
			} //end if greater than 24
		} //end for array
		
		
		//now make all the letter objects
		for (int i = 0; i < letterArray.length; i++) {
			letterArray[i] = new Letter( sec_nums[i] );
			letterArray[i].getLetter();
		} //end for letterArray
		
		//now loop through the array the other way to show all the letters in the right order
		for (int i = letterArray.length - 1; i >= 0; i--) {
			letterArray[i].showLetter();
		}
		
		System.out.print("\n");
	}
}
